features/step_definitions/kata.rb

```diff
@@ -1,229 +1,127 @@
-  step %Q|I obtain test data file "kata/release-#{cb.master_version}/subscription.yaml"|
-  channel = YAML.load(open('subscription.yaml')).dig('spec', 'channel')
-  logger.info("Subscription using channel: #{channel}")
-  cb[cb_name] = channel
+Then /^#{QUOTED} pod should run using kata runtime$/ do |pod_name|
+  step %Q|I wait until "#{pod_name}" is ready|
 end
 
-Given /^I find all pods running with kata as runtime in the cluster and store them to the#{OPT_SYM} clipboard$/ do |cb_name|
-  ensure_admin_tagged
-  cb_name ||= :kata_pods
-  @result_pods = admin.cli_exec(:get, resource: "pods", all_namespaces: true, o: "jsonpath='{.items[?(@.spec.runtimeClassName==\"kata\")].metadata.name}'")
-  @result_namespaces = admin.cli_exec(:get, resource: "pods", all_namespaces: true, o: "jsonpath='{.items[?(@.spec.runtimeClassName==\"kata\")].metadata.namespace}'")
-  pods = eval(@result_pods[:response]).split(' ')
-  ns = eval(@result_namespaces[:response]).split(' ')
-  ns_pods_list = ns.zip pods
-  kata_pods = []
-  ns_pods_list.each do |ns, p|
-    project(ns)
-    pod_obj = pod(p)
-    kata_pods << pod_obj
+Given /^I wait until #{QUOTED} pod is running$/ do |pod_name|
+  timeout = 30
+  kata_ns = "openshift-sandboxed-containers-operator"
+  expected_status = "Running"
+  pod_status = {
+    instruction: "Wait till pod is running",
+    success: false,
+  }
+  pod_status[:success] = wait_for(timeout) do
+    @result_status = admin.cli_exec(:get, resource: "pods/#{pod_name}", o: "jsonpath=''{..status.phase}'", n:kata_ns)[:stdout].to_s.include? expected_status
   end
-  cb[cb_name] = kata_pods
+  raise "Failed to deploy a pod" unless @result_status
+  logger.info("Pod deployment status is #{expected_status}")
 end
 
-Given /^I remove all kata pods in the cluster stored in the#{OPT_SYM} clipboard$/ do |cb_name|
-  cb_name ||= :kata_pods
-  cb[cb_name].each do | kp |
-    logger.info("Removing pod #{kp.name}...")
-    begin
-      res_type, res_name = kp.walk_owner_references(user: user, resource_name: pod.name, resource_type: kp)
-      unless res_type.is_a? String
-        res_type = res_type.class.name.split('BushSlicer::').last
-      end
-      resource_word = camel_to_snake_case(res_type) # BushSlicer::RESOURCES[res_type.class].to_s
-      step_sentence = "I ensure \"#{res_name}\" #{resource_word} is deleted from the \"#{project.name}\" project"
-      step %Q/#{step_sentence}/
-    rescue
-      next
-    end
+Given /^I check if #{QUOTED} runtime is kata$/ do |pod_name|
+  timeout = 30
+  kata_ns = "openshift-sandboxed-containers-operator"
+  expected_runtime = "kata"
+  pod_status = {
+    instruction: "Check if pod runtime is kata",
+    success: false,
+  }
+  pod_status[:success] = wait_for(timeout) do
+    @result_runtime = admin.cli_exec(:get, resource: "pods/#{pod_name}", o: "jsonpath=''{..spec.runtimeClassName}'", n:kata_ns)[:stdout].to_s.include? expected_runtime
```

Comment:
@valtair1891 the pod class has a method to get the runtime class. https://github.com/openshift/verification-tests/blob/master/lib/openshift/pod.rb#L234

```
pod('example-fedora-kata').runtime_class_name
```